tests/auth/providers/test_homeassistant.py

```diff
@@ -288,3 +289,18 @@ def legacy_data(hass):
             'username': 'hello '
         })
     assert credentials1 is not credentials3
+
+
+async def test_race_condition_in_data_loading(hass):
```

Comment:
I'd also suggest also testing that the calls to `self._user_store.async_load()` are called once per provider.

Reply:
More test added

Reply:
More test added